addons/sale_payment/models/payment.py

```diff
@@ -88,6 +88,8 @@ def _generate_and_pay_invoice(self):
         created_invoice = self.env['account.invoice'].browse(created_invoice).with_context(**ctx_company)
 
         if created_invoice:
+            if hasattr(self, 'account_invoice_id'):
+                self.account_invoice_id = created_invoice.id
```

Comment:
Hello :)
I am not a fan of this method, we have a modular system, we should use it as it is intended to.

From what I see, the field ```account_invoice_id``` is declared in the module "account_payment".
So we have to fix the issue in this module.

To do so, I would add a return value in the method ```_generate_and_pay_invoice``` that return either ```False``` if the invoice was not created or ```created_invoice``` if it has been created.

Then in the module "account_payment" I would override the method ```_generate_and_pay_invoice``` to correctly set the returned invoice id on the field ```account_invoice_id```.

By looking on the code, it is not dangerous nor code breaking to add a return value on ```_generate_and_pay_invoice``` since it is never overriden to add a return value.

Thanks !

Reply:
Well the issue is that there is no common inheritance point between sale_payment and account_payment ...

Reply:
Yup, I see the issue now. I still think we should fix the issue on the module that declare the field.

I would check if the parent class has the method ```_generate_and_pay_invoice```.
Since it is declared in sale_payment and never overriden this should not be dangerous.
It is crappy for sure, but crappy in the right module :P 

```python
def _generate_and_pay_invoice(self):
    """Here a super cool docstring that explains why we do this"""
    if hasattr(super(PaymentTransaction, self), '_generate_and_pay_invoice'):
        invoice_id = super(PaymentTransaction, self)._generate_and_pay_invoice()
        if invoice_id:
            self.account_invoice_id = invoice_id
```
And as I mentionned in the example right above, we should write a docstring that explains why we are overriding this method and why we check if super has the method before calling it.

Reply:
Are we sure that when computing the dependency tree sale_payment will be computed before account_payment ? As you may install sale_payment and account_payment in various installation order, I don't know how odoo will compute the dependency tree.

Reply:
Hmm true, this is not a robust fix. I have no other idea that could be clean.
The initial fix would do the work then :/

Reply:
Well we could define a base method on tx called at confirmation that does basically nothing and udpate it in account_payment. That way it is modular and code should stay short. It seems a better solution to me, what do you think about that ?

Reply:
I think this is a great idea, we should do this :+1:

Reply:
Ok, vendu  !

@nla-odoo Could you udpate your proposal according to last decision ? Which is "Well we could define a base method on tx called at confirmation that does basically nothing and udpate it in account_payment. That way it is modular and code should stay short."

Many thanks,